### PR TITLE
updated jenkinsfile to use publishNpm install of directly using npm publish

### PR DESCRIPTION
**Pull request by MichaelTzach**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,4 +1,6 @@
-// All non-runtime dependencies installed in Dockerfile.ci
+#!groovy
+@Library('ci-scripts') _
+
 pipeline {
   agent {
```
Comment (shamil) [inline Jenkinsfile:5]: Just use this line, and you won't need `Dockerfile.ci` file

```suggestion
  agent any
```

File: Jenkinsfile [hunk 2]
```diff
@@ -11,9 +13,13 @@ pipeline {
       }
 
       steps {
-        withCredentials([string(credentialsId: 'npm-bringg', variable: 'NPM_TOKEN')]) { writeFile file: '.npmrc', text: "//registry.npmjs.org/:_authToken=${env.NPM_TOKEN}" }
+        // package.json should not have private dependencies
+        // we should be able to install dependencies without credentials
         sh 'yarn'
```
Comment (yosiat) [inline Jenkinsfile:18]: we are not using yarn at bringg, can you change it to be npm?

File: Jenkinsfile [hunk 3]
```diff
@@ -1,4 +1,6 @@
-// All non-runtime dependencies installed in Dockerfile.ci
+#!groovy
```
Comment (yosiat) [inline Jenkinsfile:1]: it looks like you are using jenkins only for deployment, what about running tests? making sure the typescript code is compiled? prettier  is passing? etc ..

File: Jenkinsfile (hunk 2 above)
Comment (MichaelTzach) [reply Jenkinsfile:18]: changed

File: Jenkinsfile (hunk 3 above)
Comment (MichaelTzach) [reply Jenkinsfile:1]: I will add those in a later stage.

The tests are just going to be making sure that typescript code is compiles as there is no good way that I could find for making sure that the native code is being used correctly.

What we are going to do for e2e tests is have this run in automation.

File: Jenkinsfile (hunk 1 above)
Comment (MichaelTzach) [reply Jenkinsfile:5]: changed

File: Jenkinsfile (hunk 3 above)
Comment (yosiat) [reply Jenkinsfile:1]: not sure why not to add check compilation check, but ok..

**Pull request merged**